Source/WebCore/platform/network/ResourceResponseBase.cpp

```diff
@@ -131,13 +132,13 @@ ResourceResponse ResourceResponseBase::fromCrossThreadData(CrossThreadData&& dat
     ResourceResponse response;
 
     response.setURL(data.url);
-    response.setMimeType(AtomString { data.mimeType });
+    response.setMimeType(AtomString { WTFMove(data.mimeType).isolatedCopy() });
```

Comment:
I admit I don't understand this change. The whole point of CrossThreadData is that it should contain strings that have already been isolated. It seems therefore odd to require the recipient of said CrossThreadData to isolate copy those strings again. I have a suspicion there must be a better fix.

Reply:
Like thread A calls r.crossThreadData() and captures this in a lambda when it dispatches to thread B.
Thread B should be able to create a response from this CrossThreadData without further isolate copying, since this was already done by the crossThreadData() on thread A. So in the general case, it should not be needed but your patch adds this cost anyway.

I suspect your use case if doing something more complicated that is slightly wrong and we should be fixing that wrong thing instead of updating fromCrossThreadData(). E.g. Maybe we have code that dispatches the CrossThreadData yet again to another thread, without isolatedCopying() it?

Reply:
The issue is that `AtomString(a)`, `a` being isolatedCopy will change the StringImpl of `a` and it will no longer be isolatedCopy. With a WorkQueue that might dispatch tasks in different threads, we end up into issues.

I'll change ResourceResponseBase::CrossThreadData::copy() instead of fromCrossThreadData though, this seems cleaner this way.

Reply:
We have a work queue.
We dispatches a first task on this queue where we create a ResourceResponse from a CrossThreadData that is stored in a record.
At that point, the record's CrossThreadData String will be changed to an AtomString.
The same work queue dispatches a second task where we remove the record, hence we destroy the CrosssThreadData. This triggers the removal of the AtomString but from a different thread it was created on.